File: be/shopapp/src/main/java/com/project/shopapp/models/Article.java
package com.project.shopapp.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.validator.constraints.Length;

@Entity
@Table(name = "articles")
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Article extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Length(max = 3000)
    private String description;

    private String thumbnail;

    private String author;

    @Lob
    private String content;
}

File: fe/src/app/models/article.ts
export interface Article {
  id: number;
  author: string;
  title: string;
  description: string;
  thumbnail: string;
  created_at: Date;
  content: string;
  url: string;
}

export interface UpdateArticle {
  author: string;
  title: string;
  description: string;
  thumbnail: string;
  created_at: Date;
}

export interface InsertArticle {
  author: string;
  title: string;
  description: string;
  thumbnail: string;
}


